source/Agents/RandomAgent.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * This file is part of the Fall 2023, CSE 491 Course Project
+ * @file RandomAgent.hpp
+ * @brief An Agent that will move around using random actions
+ * @note Status: PROPOSAL
+ * @author Yousif Murrani
+ **/
+
+#pragma once
+
+#include <cassert>
+#include "../core/AgentBase.hpp"
+#include "../core/GridPosition.hpp"
+#include "../core/WorldBase.hpp"
+
+namespace walle {
+
+    /**
+     * Class for the Random Agent
+     */
+    class RandomAgent : public cse491::AgentBase {
+
+        private:
+
+            double random = 4.0; /// Random value used to determine direction
```

Comment:
Probably okay for now, but this variable name is a little confusing. 
I've almost always seen `random` being used for the actual random number generator. If this is just a multiplier / limits / whatever, the name could specify that.